doc/MAPGEN.md

```diff
@@ -1386,3 +1389,143 @@ update_mapgen adds new optional keywords to a few mapgen JSON items.
 place_npc, place_monster, and place_computer can take an optional target boolean. If they have `"target": true` and are
 invoked by update_mapgen with a valid mission, then the NPC, monster, or computer will be marked as the target of the
 mission.
+
+# Map Extras
+
+Map extras can be used to place environmental objects and structures that can help create some emergent storytelling. These are placed randomly while exploring the overmap, and can range from simple ponds and groves to full-on crashed spaceships with enemies or NPCs.
+
+### "map_extra"
+
+```JSON
+{
+  "id": "mx_minefield",
+  "type": "map_extra",
+  "name": { "str": "Minefield" },
+  "description": "Mines are scattered here.",
+  "generator": { "generator_method": "map_extra_function", "generator_id": "mx_minefield" },
+  "min_max_zlevel": [ 0, 0 ],
+  "sym": "M",
+  "color": "red",
+  "autonote": true,
+  "flags": [ "MAN_MADE" ]
+}
+```
+
+| Field          | Description
+| ---            | ---
+| generator      | (_optional_) An object defining how and what this extra should spawn.
```

Comment:
I thought this field is mandatory?

Reply:
You can actually omit this field, it will just use a generator_method of `null` (might be useful for overriding the map extra, otherwise not much utility)